drivers/overlay/overlay.go

```diff
@@ -1666,9 +1666,11 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 	}
 
 	mergedDir := path.Join(dir, "merged")
-	// Create the driver merged dir
-	if err := idtools.MkdirAs(mergedDir, 0o700, rootUID, rootGID); err != nil && !os.IsExist(err) {
-		return "", err
+	// Attempt to create the merged dir only if it doesn't exist.
+	if _, err := os.Stat(mergedDir); err != nil && !os.IsExist(err) {
```

Comment:
```suggestion
	if _, err := os.Stat(mergedDir); err != nil && os.IsExist(err) {
```

Reply:
ah, it should really be `os.IsNotExist(err)`

Reply:
And you are not supposted to use IsExist any longer, `errors.Is(err, fs.ErrNotExist)` But it looks like we have some cases of IsNotExist in podman.